tests/testthat/test-chop.R

```diff
@@ -100,33 +130,113 @@ test_that("correctly performs tidy recycling with size 1 inputs", {
   expect_identical(unchop(df, c(x, y)), expect)
 })
 
-test_that("can specify a ptype with extra columns", {
+test_that("nonexistent `ptype` columns are ignored", {
   df <- tibble(x = 1, y = list(1, 2))
-  ptype <- tibble(y = numeric(), z = numeric())
-
-  expect <- tibble(x = c(1, 1), y = c(1, 2), z = c(NA_real_, NA_real_))
+  ptype <- list(y = numeric(), z = numeric())
 
-  expect_identical(unchop(df, y, ptype = ptype), expect)
+  expect_identical(unchop(df, y, ptype = ptype), unchop(df, y))
 })
```

Comment:
This is the weird `ptype` behavior that we used to have. Notice that `z` was a column that only existed in the `ptype`, but that forced it to exist in the `unchop()` result as an all missing column. That seemed pretty silly, so I've removed that behavior.